instrumentation/r2dbc-1.0/library/src/main/java/io/opentelemetry/javaagent/instrumentation/r2dbc/v1_0/TraceProxyListener.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.r2dbc.v1_0;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.javaagent.instrumentation.r2dbc.v1_0.internal.DbExecution;
+import io.r2dbc.proxy.core.QueryExecutionInfo;
+import io.r2dbc.proxy.listener.ProxyMethodExecutionListener;
+import io.r2dbc.spi.ConnectionFactoryOptions;
+
+public class TraceProxyListener implements ProxyMethodExecutionListener {
```

Comment:
This class could be made final. I think it should also be moved to the `internal` package (or made package private).